php/class-wp-seo-social-settings.php

```diff
@@ -0,0 +1,708 @@
+				'args'     => array(
+					'field' => 'archive_date_og_description',
+					'type'  => 'textarea',
+				),
+			),
+			array(
+				'id'       => 'archive_date_og_image',
+				'title'    => __( 'Open Graph Image', 'wp-seo-social' ),
+				'section'  => 'archive_date',
+				'args'     => array(
+					'field' => 'archive_date_og_image',
+					'type'  => 'image',
+				),
+			),
+			array(
+				'id'       => 'archive_date_og_type',
+				'title'    => __( 'Open Graph Type', 'wp-seo-social' ),
+				'section'  => 'archive_date',
+				'args'     => array(
+					'field' => 'archive_date_og_type',
+					'type'  => 'dropdown',
+					'boxes' => array(
+						'website' => 'Website',
+						'article' => 'Article',
+					),
+				),
+			),
+			array(
+				'id'       => 'search_og_title',
+				'title'    => __( 'Open Graph Title Tag Format', 'wp-seo-social' ),
+				'section'  => 'search',
+				'args'     => array(
+					'field' => 'search_og_title',
+				),
+			),
+			array(
+				'id'       => 'search_og_description',
+				'title'    => __( 'Open Graph Description Tag Format', 'wp-seo-social' ),
+				'section'  => 'search',
+				'args'     => array(
+					'field' => 'search_og_description',
+					'type'  => 'textarea',
+				),
+			),
+			array(
+				'id'       => 'search_og_image',
+				'title'    => __( 'Open Graph Image', 'wp-seo-social' ),
+				'section'  => 'search',
+				'args'     => array(
+					'field' => 'search_og_image',
+					'type'  => 'image',
+				),
+			),
+			array(
+				'id'       => 'search_og_type',
+				'title'    => __( 'Open Graph Type', 'wp-seo-social' ),
+				'section'  => 'search',
+				'args'     => array(
+					'field' => 'search_og_type',
+					'type'  => 'dropdown',
+					'boxes' => array(
+						'website' => 'Website',
+						'article' => 'Article',
+					),
+				),
+			),
+			array(
+				'id'       => '404_og_title',
+				'title'    => __( 'Open Graph Title Tag Format', 'wp-seo-social' ),
+				'section'  => '404',
+				'args'     => array(
+					'field' => '404_og_title',
+				),
+			),
+			array(
+				'id'       => '404_og_description',
+				'title'    => __( 'Open Graph Description Tag Format', 'wp-seo-social' ),
+				'section'  => '404',
+				'args'     => array(
+					'field' => '404_og_description',
+					'type'  => 'textarea',
+				),
+			),
+			array(
+				'id'       => '404_og_image',
+				'title'    => __( 'Open Graph Image', 'wp-seo-social' ),
+				'section'  => '404',
+				'args'     => array(
+					'field' => '404_og_image',
+					'type'  => 'image',
+				),
+			),
+			array(
+				'id'       => '404_og_type',
+				'title'    => __( 'Open Graph Type', 'wp-seo-social' ),
+				'section'  => '404',
+				'args'     => array(
+					'field' => '404_og_type',
+					'type'  => 'dropdown',
+					'boxes' => array(
+						'website' => 'Website',
+						'article' => 'Article',
+					),
+				),
+			),
+		);
+		$taxonomy_settings = array();
+		foreach ( $this->taxonomies as $taxonomy ) {
+			$taxonomy_settings = array_merge(
```

Comment:
Maybe I am just not understanding this code, but why can't you just do
```php
taxonomy_settings[] = array();
```
instead of trying to merge it with itself each time?  The same goes for all of the below loops as well.